runtime/runtime.go

```diff
@@ -147,11 +147,19 @@ func list() []string {
 	return ret
 }
 
-func register(failpath string, fp *Failpoint) {
+func register(name string) *Failpoint {
 	failpointsMu.Lock()
-	failpoints[failpath] = fp
-	failpointsMu.Unlock()
-	if t, ok := envTerms[failpath]; ok {
-		Enable(failpath, t)
+	defer failpointsMu.Unlock()
+	if _, ok := failpoints[name]; ok {
+		panic(fmt.Sprintf(
+			"failpoint name %s is already registered. You are using same failpoint name in multiple packages.\n",
```

Comment:
```suggestion
			"failpoint name %s is already registered.",
```
Could be same package.